File: 58.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        s = s.strip()
        if len(s) == 0:
            return 0
        if " " not in s:
            return len(s)
        strLen = len(s)
        i = strLen - 1
        while s[i] != " " and i > 0:
            i -= 1
        return strLen - i - 1

File: 58.js
// First solution
const lengthOfLastWord = (s) => {
	s = s.trim();
	const strLength = s.length;
	const space = " ";
	if (s.indexOf(space) === -1) {
		return strLength;
	}
	for (let i = strLength - 1; i >= 0; i--) {
		if (s[i] === space) {
			return strLength - i - 1;
		}
	}
};

console.log(lengthOfLastWord("Hello World")); // 5
console.log(lengthOfLastWord("   fly me   to   the moon  ")); // 4
console.log(lengthOfLastWord("luffy is still joyboy")); // 6

// Second solution
const lengthOfLastWord_1 = (s) => s.trim().split(" ").at(-1).length;
